cinema/views.py

```diff
@@ -1,45 +1,113 @@
-from rest_framework.decorators import api_view
+from rest_framework import generics, viewsets, status, mixins
+from django.shortcuts import get_object_or_404
+
 from rest_framework.response import Response
-from rest_framework import status
+from rest_framework.views import APIView
 
-from django.shortcuts import get_object_or_404
+from cinema.models import Movie, Actor, Genre, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+
+class ActorList(
+    generics.ListAPIView,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request, *args, **kwargs):
+        return self.list(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        return self.create(request, *args, **kwargs)
+
+
+class ActorDetail(
+    generics.GenericAPIView,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.DestroyModelMixin,
+):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
+    def get(self, request, *args, **kwargs):
+        return self.retrieve(request, *args, **kwargs)
 
+    def put(self, request, *args, **kwargs):
+        return self.update(request, *args, **kwargs)
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
+    def patch(self, request, *args, **kwargs):
+        return self.partial_update(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        return self.destroy(request, *args, **kwargs)
+
+
+class GenreList(APIView):
```

Comment:
The `GenreList` class could be simplified by using `ListCreateAPIView` instead of `APIView`. This would automatically handle both `GET` and `POST` requests without needing to manually define these methods.